ocaml-backend/src/main/java/org/kframework/backend/ocaml/DefinitionToOcaml.java

```diff
@@ -453,29 +454,32 @@ public String interpreter() {
         ocamlProgramHeader(sb, true);
         ocamlMatchPattern(exitCodePattern, sb);  //declares try_match
         sb.append("let input, depth, out = Makeconfig.parse ()\n");
-        sb.append("let res, _ = ");
-        if (this.threadCellExists) {
-            sb.append("run");
-        } else {
-            sb.append("run_no_thread_opt");
-        }
-        sb.append("(input) (depth)\n");
+        sb.append("let stack, _ = ");
+        constructRun(sb, "depth");
+        ocamlPrintKStacktrace(sb);
+        sb.append("let res = hd (rev stack)\n");
         sb.append("let () = output_string out (print_k_binary(res))\n");
         sb.append("let () = try let subst = try_match res res (-1) in\n");
         sb.append("let code = get_exit_code subst in\n");
         sb.append("exit code\n");
-        sb.append("with Stuck(res) -> output_string out (print_k_binary res); exit 112\n");
+        sb.append("with Stuck(stack2) -> output_string out (print_k_binary (hd (rev stack2))); exit 112\n");
         return sb.toString();
     }
 
     private void matchPatternAndPrint(StringBuilder sb) {
         sb.append("let _ = (try print_subst_binary subst (try_match res res (-1)) with Stuck c -> output_string subst \"\\x00\\x00\\x00\\x00\")\n");
     }
 
+    private void ocamlPrintKStacktrace(StringBuilder sb) {
+		sb.append("let () = if (length stack) <> 1 then print_string (print_stacktrace stack) \n");
```

Comment:
I'm not sure I'm really comfortable with this being dumped on stdout without any kind of control happening here. It breaks the architecture of krun to be printing stuff to stdout like this. I'd feel a lot better if the terms were serialized to the java frontend and pretty printed there.